binstring/binstring.ml

```diff
@@ -0,0 +1,125 @@
+module Str = Stdlib.String
+module Buf = Stdlib.Buffer
+
+type t = { inner : string list; offset : int; length : int }
```

Comment:
This internal representation can change later, right?

Reply:
100% – this is an abstract type from the outside.

Reply:
done!